### PR TITLE
Fix language preferentiality for duplicate resources

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary
* Fixes historically incorrect language code for Chewa, we had previously set it as 'nyn' which is for a completely different language.
* Adds a migration step to migrate device settings from 'nyn' to the actual language code 'ny'
* Fixes previously unnoticed regression where the copies modal for duplicate resources would not work
* Adds support for cross checking multiple language codes to see if the current UI language matches the content language

## References
Fixes [#11751](https://github.com/learningequality/kolibri/issues/11751)

## Reviewer guidance
* Change the user interface to Chewa and ensure that it is still translated.
* Run 0.16.1, select Chewa as the default language. Upgrade to the asset from this PR and ensure that it is properly still showing in Chewa by default.
* Install the first topic of KA English and KA Fufulde.
* Engage with one exercise in Fufulde, do not master it.
* Go to the home page, confirm that it appears in the resume learning in English if the UI is in English, check that the '2 locations' is clickable and the modal works.
* Change the UI language to Fufulde, confirm that the resume learning shows the resource in Fufulde now.

Import duplicate resources in the KA English channel. Search for them, confirm that the duplicate resource modal works there as well.

----

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

----

| :rocket: This description was created by [Ellipsis](https://www.ellipsis.dev) for commit 2d074f33d46415e6dc4aa176d226418fc53b0bdf  | 
|--------|

### Summary:
This PR corrects the Chewa language code, adds necessary migrations, updates handling of duplicate resources, and enhances language matching logic.

**Key points**:
- Corrects Chewa language code from `nyn` to `ny`.
- Adds migration for updated language code in device settings.
- Updates duplicate resource modal handling in various views.
- Enhances language matching logic in `getContentLangActive` function.
- Ensures UI consistency and functionality post-migration.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

**Comment by rtibbles** [comment]
Hey @ellipsis, give me a code review

File: kolibri/core/assets/src/utils/i18n.js [hunk 1]
```diff
@@ -25,6 +25,27 @@ export const getContentLangDir = language => {
...
 
+const contentLanguageCodes = {
+  ff: ['ful', 'fuv'],
+  ny: ['nya'],
+  sw: ['swa'],
+  yo: ['yor'],
+};
+
+export const getContentLangActive = language => {
+  const langCode = languageIdToCode(currentLanguage);
```
[the remark this answers is not in the document]
Comment (rtibbles) [reply kolibri/core/assets/src/utils/i18n.js:36]: It's defined in the module scope.

**Comment by radinamatic** [comment]
No luck with seeing the Chewa UI after upgrading from 0.16.1 to asset from this PR. ``nyn`` locale is not found (which was the point, I believe), but neither is any trace of the Chewa language in the language switcher, and the default device language is returned to browser. If the locale is manually edited to ``ny``, it returns 404.

https://github.com/learningequality/kolibri/assets/1457929/6deea8f6-45aa-4ed7-ba38-04c2aa5916b6

 Tested on Ubuntu 20.04 with Firefox and Chrome.

**Comment by rtibbles** [comment]
Thanks @radinamatic - I am pretty sure I have forgotten to update the language code in one place. I'll do a quick search for the whole codebase for `nyn` to double check!

**Comment by rtibbles** [comment]
There were a couple of spots I had missed. Including the fact that Intl.js doesn't actually have a polyfill for Chewa locale data. Thankfully, I was able to generate one automatically from sources. I think it should all be working as intended now.

**Comment by radinamatic** [comment]
After the upgrade from 0.16.1 using the newest asset in this PR, `nyn` locale that was previously set as default logically gives the 404 error, but going to the home page reloads the new `ny` locale, and the Chewa language is present in the language switcher modal.

https://github.com/learningequality/kolibri/assets/1457929/d55902b9-aeb5-47a4-a511-ed5c9732a61b

All is also well after installing the first topic of KA English and KA Fufulde, exercises engaged and not mastered appear in the _resume..._ section, and the '2 locations' link is clickable and the modal works in both locales.

Fulfulde | English
-- | --
babe | locations-1
ff-cm | en


Importing duplicate resources in the KA English channel also displays the 2 locations link and the duplicate resource modal works

locations

**Review by radinamatic — APPROVED** [review]
Manual QA passes, good to go! 👏🏽 💯 :shipit:

**Review by marcellamaki — APPROVED** [review]
Code review looks good to me. I think the sorting situation with the matching language codes is the better option for handling a bit of odd perfect/acceptable matching, and the comments are helpful to make it more parseable for other use cases.

Remains to be seen if future Marcella agrees with me.

**Pull request merged**